### ISSUE TITLE
some SPDX json generated from Syft doesn't include DocumentName 

### ISSUE THREAD

**Issue by lumjjb** [guacsec/guac#743]
Some SPDX json generated from Syft doesn't include DocumentName. This is not in accordance to the SPDX spec (https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#64-document-name-field)...

Unfortunately, I am not able to share these SBOMs, but they were generated by Syft.

**Comment by JonZeolla** [comment]
I ran into a similar issue today where my spdx-json file had a name of `./` which resulted in a segfault:

```bash
$ bin/guacone collect files ./spdx.json
{"level":"info","ts":1687301853.469325,"caller":"cli/init.go:53","msg":"Using config file: /Users/jonzeolla/src/jonzeolla/guac/guac.yaml"}
{"level":"info","ts":1687301853.6743119,"caller":"parser/parser.go:126","msg":"parsing document tree with root type: SPDX"}
{"level":"error","ts":1687301853.750566,"caller":"collector/collector.go:89","msg":"emit error: unable to ingest doc tree: unable to parse purl pkg:guac/spdx/./: name is required","stacktrace":"github.com/guacsec/guac/pkg/handler/collector.Collect\n\t/Users/jonzeolla/src/jonzeolla/guac/pkg/handler/collector/collector.go:89\ngithub.com/guacsec/guac/cmd/guacone/cmd.glob..func5\n\t/Users/jonzeolla/src/jonzeolla/guac/cmd/guacone/cmd/files.go:181\ngithub.com/spf13/cobra.(*Command).execute\n\t/Users/jonzeolla/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/Users/jonzeolla/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068\ngithub.com/spf13/cobra.(*Command).Execute\n\t/Users/jonzeolla/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992\ngithub.com/guacsec/guac/cmd/guacone/cmd.Execute\n\t/Users/jonzeolla/src/jonzeolla/guac/cmd/guacone/cmd/root.go:51\nmain.main\n\t/Users/jonzeolla/src/jonzeolla/guac/cmd/guacone/main.go:23\nruntime.main\n\t/opt/homebrew/opt/go/libexec/src/runtime/proc.go:250"}
```

I worked around it by manually updating the name:

```bash
$ diff spdx.json spdx-modified.json
5c5
<  "name": "./",
---
>  "name": "example",
```

This seems to be on the syft backlog per https://github.com/anchore/syft/issues/1238

**Comment by lumjjb** [comment]
Resolved with #744
